test-helpers/src/data/encoded_value.rs

```diff
@@ -0,0 +1,61 @@
+use serde::{Deserialize, Serialize};
+
+use crate::data::TestError;
+
+// Supported value encodings:
+pub const ENCODING_BASE_64: &str = "base64";
+pub const ENCODING_HEX: &str = "hex";
+pub const ENCODING_UTF8: &str = "utf-8";
```

Comment:
yaml supports enums as `!Tag` if you wanted to use unit enums instead of strings.

Reply:
It looks like serde yaml supports enums out of the box! It appears to translate the enum variants to strings natively. I don't need to use these consts or even the `!Tag` syntax. The updated code looks like this:

```rust
#[derive(Serialize, Deserialize, Clone)]
#[serde(rename_all = "lowercase")]
pub enum Encoding {
    Base64,
    Hex,
    #[serde(rename = "utf-8")]
    Utf8,
}

#[derive(Serialize, Deserialize, Clone)]
pub struct EncodedValue {
    value: String,
    encoding: Encoding,
}
```

This simplifies the code nicely.